### PR TITLE
+ | Add StatifyBot Website to Presences

### PR DESCRIPTION
**Pull request by tayron1**
[1. Screenshot](https://why-do-you-look.nothing-to-see-he.re/3gbZAD)
[2. Screenshot](https://why-do-you-look.nothing-to-see-he.re/F5gpf8)

### CODE REVIEW

File: websites/S/StatifyBot/dist/metadata.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    "service": "StatifyBot",
+    "altnames": ["Statify", "StatifyBot", "Statify Bot", "Statify Dashboard", "Statify Website"],
+    "description": {
+      "en": "Statify is an advanced Statistic Discord Bot with many features like Charts, Leaderboards and more.",
+      "de": "Statify ist ein forgeschrittener Statistiken Discord Bot mit vielen Funktionen wie Diagramme, Bestenlisten und mehr."
+    },
+    "url": ["dev.statifybot.net", "statifybot.net", "dash.statifybot.net"],
+    "version": "0.0.1",
+    "logo": "https://visionvenue.de/PicsVV/StatifyBotV3.png",
+    "thumbnail": "https://visionvenue.de/PicsVV/StatifyBanner.png",
```
Comment (Bas950) [inline websites/S/StatifyBot/dist/metadata.json:22]: upload to imgur

File: websites/S/StatifyBot/presence.ts [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    while (c.charAt(0) == " ") {
+      c = c.substring(1);
+    }
+    if (c.indexOf(name) == 0) {
+      return c.substring(name.length, c.length);
+    }
+  }
+  return "";
+}
+
```
Comment (Bas950) [inline websites/S/StatifyBot/presence.ts:40]: ????

File: websites/S/StatifyBot/presence.ts [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+presence.on("UpdateData", async () => {
+
+
+  // Create presence data
+  const presenceData: PresenceData = {
+    largeImageKey: "statify"
+  };
+
+  const subdomain = getSubdomain();
+  const path = window.location.pathname;
```
Comment (Bas950) [inline websites/S/StatifyBot/presence.ts:52]: merge consts

File: websites/S/StatifyBot/presence.ts [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+  presenceData.smallImageText = presenceData.details.replace("...", " the") + " " + presenceData.state;
+  presenceData.buttons = [
+    {
+      label: "Visit Website",
+      url: "https://statifybot.net" + path 
+    },
+    {
+      label: "Invite Statify",
+      url: "https://statifybot.net/invite"
+    }
```
Comment (Bas950) [inline websites/S/StatifyBot/presence.ts:139]: that one aint allowed, as its an advertisement.

File: websites/S/StatifyBot/presence.ts [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+    presenceData.details = "Viewing...";
+    presenceData.state = "Statify Designs";
+    presenceData.smallImageKey = "file";
+  } else if (path.startsWith("/premium")) {
+    presenceData.details = "Viewing...";
+    presenceData.state = "Features of Statify Premium";
+    presenceData.smallImageKey = "giveaway";
+  } else if (path.startsWith("/commands")) {
+    var url_string = window.location.toString()
+    var url = new URL(url_string);
```
Comment (Bas950) [inline websites/S/StatifyBot/presence.ts:87]: no var

**Review by Bas950 — CHANGES_REQUESTED** [review]

**Comment by ririxidev** [comment]
Sorry but this site is probably work-in-progress I guess so we are not going to accept it unfortunately.

**Comment by tayron1** [comment]
> Sorry but this site is probably work-in-progress I guess so we are not going to accept it unfortunately.

The Site is getting Updates all the time

File: websites/S/StatifyBot/dist/metadata.json [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    },
+    "contributors": [
+      {
+        "name": "Tayron",
+        "id": "510104921887211533"
+      }
+    ],
+    "service": "StatifyBot",
+    "altnames": ["Statify", "StatifyBot", "Statify Bot", "Statify Dashboard", "Statify Website"],
+    "description": {
```
[the remark this answers is not in the document]
Comment (tayron1) [reply websites/S/StatifyBot/dist/metadata.json:15]: Do you mean the Description or what do you mean?

**Comment by tayron1** [comment]
Because its a always updating site it is always work-in-progress. The only thing that is a Placeholder is the Demo vid on the Main Page because our Trailer is still in progress

**Pull request closed without merging**